src/integration/asset/iothink/IothinkAssetIntegration.ts

```diff
@@ -65,24 +65,67 @@ export default class IothinkAssetIntegration extends AssetIntegration<AssetSetti
 
   private filterConsumptionRequest(asset: Asset, data: any, manualCall: boolean): AbstractCurrentConsumption[] {
     const consumptions: AbstractCurrentConsumption[] = [];
+    // Rename the value property in logs with tag reference
+    for (const dataSet of data.historics) {
+      dataSet.logs.forEach((log) => {
+        log[dataSet.tagReference] = log['value'];
+        delete log['value'];
+      });
+    }
+    // Create helper map to merge the arrays to combined objects
+    const map = new Map();
+    for (const dataSet of data.historics) {
+      dataSet.logs.forEach((item) => map.set(item.timestamp, { ...map.get(item.timestamp), ...item }));
+    }
+    // Create Array with merged consumptions from map
+    const mergedResponse = Array.from(map.values());
+    // Sort the Array according the timestamp
+    mergedResponse.sort((a, b) => a.timestamp - b.timestamp);
     const energyDirection = asset.assetType === AssetType.PRODUCTION ? -1 : 1;
-    if (!Utils.isEmptyArray(data.historics)) {
-      for (let i = 0; i < data.historics[0].logs.length; i++) {
+    if (!Utils.isEmptyArray(mergedResponse)) {
+      for (const mergedConsumption of mergedResponse) {
```

Comment:
Add a type of mergedResponse?

Reply:
A type of the mergedResponse would look like the property enum with an added timestamp: 
```
export interface IothinkMergedResponse {
  timestamp: number;
  Puissance_active_de_phase_1?: string;
  Puissance_active_de_phase_2:? string;
  Puissance_active_de_phase_3:? string;
  'io-pow-active'?: string;
  'io-energy-input'?: string;
  'io-energy-output'?: string;
  'io-energy'?: string;
  'io-energy-charge'?: string;
  'io-energy-discharge'?: string;
  'io-soc'?: string;
}
```
If we do it like this, the enum is obsolete and can be removed.

Reply:
Okay.
But we have now just 3 weird namings here, could we separate them?

Reply:
This is how the property names of the response are called. 
That is why I used the enum to get the property values from the merged Array response. Do you want to separate the different namings in three different types or do you want blank lines between the different namings?

Reply:
If it matches the ioThink endpoint, let's keep it as it is then.